### PR TITLE
Add rabbitmq user type and controller

### PR DESCRIPTION
**Pull request by mehtapratikd**

### CODE REVIEW

File: pkg/controller/rabbitmquser/controller.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/Ridecell/ridecell-operator/pkg/components"
+	rmqucomponents "github.com/Ridecell/ridecell-operator/pkg/controller/rabbitmquser/components"
+)
+
+// Add creates a new DjangoUser Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	_, err := components.NewReconciler("rabbitmq-user-controller", mgr, &dbv1beta1.RabbitmqUser{}, nil, []components.Component{
+		rmqucomponents.NewDefaults(),
```
Comment (coderanger) [inline pkg/controller/rabbitmquser/controller.go]: Missing the user component (and any other you add).

File: pkg/controller/summon/components/app_secrets.go [hunk 2]
```diff
@@ -127,13 +126,14 @@ func (comp *appSecretComponent) Reconcile(ctx *components.ComponentContext) (com
...
 	}
-
 	// This order must match the one in inputSecrets().
 	postgresSecret := dynamicInputSecrets[0]
 	fernetKeys := dynamicInputSecrets[1]
 	secretKey := dynamicInputSecrets[2]
 	awsSecret := dynamicInputSecrets[3]
+	rabbitmqPassword := dynamicInputSecrets[4]
 
+	fmt.Println(rabbitmqPassword)
```
Comment (coderanger) [inline pkg/controller/summon/components/app_secrets.go]: Make sure this gets removed before merge.

File: pkg/controller/summon/components/app_secrets.go [hunk 3]
```diff
@@ -127,13 +126,14 @@ func (comp *appSecretComponent) Reconcile(ctx *components.ComponentContext) (com
 	if err != nil {
 		return components.Result{Requeue: true}, err
 	}
-
```
Comment (coderanger) [inline pkg/controller/summon/components/app_secrets.go]: Don't remove whitespace that isn't part of your change?

File: pkg/controller/summon/components/app_secrets.go [hunk 4]
```diff
@@ -127,13 +126,14 @@ func (comp *appSecretComponent) Reconcile(ctx *components.ComponentContext) (com
...
 	// This order must match the one in inputSecrets().
 	postgresSecret := dynamicInputSecrets[0]
 	fernetKeys := dynamicInputSecrets[1]
 	secretKey := dynamicInputSecrets[2]
 	awsSecret := dynamicInputSecrets[3]
+	rabbitmqPassword := dynamicInputSecrets[4]
```
Comment (coderanger) [inline pkg/controller/summon/components/app_secrets.go]: Still need to actually use this in the app secret.

File: pkg/controller/rabbitmquser/controller.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (coderanger) [inline pkg/controller/rabbitmquser/controller.go]: Missing the Secret component.

File: pkg/controller/rabbitmquser/components/secrets.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		err = ctx.Create(ctx.Context, target)
+	}
+	if err != nil {
+		return components.Result{Requeue: true}, errors.Wrapf(err, "secret: unable to save secret %s/%s", instance.Namespace, secretName)
+	}
+
+	return components.Result{}, nil
+}
+
+func RandStringRunes(n int) string {
```
Comment (coderanger) [inline pkg/controller/rabbitmquser/components/secrets.go]: As an internal function, this should generally be private, meaning `randStringRunes`.

File: pkg/apis/db/v1beta1/rabbitmquser_types.go [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RabbitmqUserSpec defines the desired state of RabbitmqUser
+type RabbitmqUserSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	Username   string             `json:"username"`
+	Tags       string             `json:"tags"`
+	Connection RabbitmqConnection `json:"connection"`
```
Comment (coderanger) [inline pkg/apis/db/v1beta1/rabbitmquser_types.go]: We still may need to revisit this method of storing the connection info, need to figure out the best way to manage the operator-level credentials.

File: pkg/controller/summon/components/app_secrets.go (hunk 3 above)
Comment (mehtapratikd) [reply pkg/controller/summon/components/app_secrets.go]: Yup ! Agree

**Comment by coderanger** [comment]
Replaced by #63

**Pull request closed without merging**